src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -111,7 +114,18 @@ protected void runTool(final JavaSparkContext ctx) {
         //NOTE: this doesn't honor enabled/disabled commandline filters
         final ReadFilter bqsrReadFilter = ReadFilter.fromList(BaseRecalibrator.getBQSRSpecificReadFilterList(), getHeaderForReads());
 
-        final JavaRDD<GATKRead> markedFilteredReadsForBQSR = markedReads.filter(read -> bqsrReadFilter.test(read));
+        JavaRDD<GATKRead> markedFilteredReadsForBQSR = markedReads.filter(read -> bqsrReadFilter.test(read));
+
+        if (joinStrategy.equals(JoinStrategy.OVERLAPS_PARTITIONER)) {
+            // the overlaps partitioner requires that reads are coordinate-sorted
+            final SAMFileHeader readsHeader = getHeaderForReads().clone();
+            readsHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+            ReadCoordinateComparator comparator = new ReadCoordinateComparator(readsHeader);
+            markedFilteredReadsForBQSR = markedFilteredReadsForBQSR
+                    .mapToPair(read -> new Tuple2<>(read, null))
+                    .sortByKey(comparator, true, numReducers)
+                    .keys();
+        }
```

Comment:
Note that the reads are unsorted after MD, so we need to make them coordinate-sorted for the overlaps partitioner. This incurs another shuffle, so we'll see how this performs (vs broadcast for example), when we benchmark longer pipelines.